db/db_versioned.go

```diff
@@ -38,23 +41,28 @@ type (
 		// Delete deletes a record by (namespace, key)
 		Delete(uint64, string, []byte) error
 
-		// Base returns the underlying KVStore
-		Base() KVStore
+		// Filter returns <k, v> pair in a bucket that meet the condition
+		Filter(string, Condition, []byte, []byte) ([][]byte, [][]byte, error)
+
+		// AddVersionedNamespace adds a versioned namespace
+		AddVersionedNamespace(string, uint32) error
```

Comment:
This seems more suitable as an initialization setting, and we seemingly do not need to support dynamic settings.

Reply:
it will be called later in DB initialization